lib/features/auth/view/screens/sign_up_screen.dart

```diff
@@ -0,0 +1,210 @@
+  final FocusNode emailFocusNode = FocusNode();
+  final FocusNode passwordFocusNode = FocusNode();
+  final FocusNode confirmPasswordFocusNode = FocusNode();
+  bool isEmailFocused = false;
+  bool isPasswordFocused = false;
+  bool isConfirmPasswordFocused = false;
+
+  final TextEditingController emailController = TextEditingController();
+  final TextEditingController passwordController = TextEditingController();
+  final TextEditingController confirmPasswordController =
+      TextEditingController();
+
+  final emailShakeKey = GlobalKey<ShakeWidgetState>();
+  final passwordShakeKey = GlobalKey<ShakeWidgetState>();
+  final confirmPasswordShakeKey = GlobalKey<ShakeWidgetState>();
+
+  @override
+  void initState() {
+    super.initState();
+    emailFocusNode.addListener(() {
+      setState(() {
+        isEmailFocused = emailFocusNode.hasFocus;
+      });
+    });
+    passwordFocusNode.addListener(() {
+      setState(() {
+        isPasswordFocused = passwordFocusNode.hasFocus;
+      });
+    });
+    confirmPasswordFocusNode.addListener(() {
+      setState(() {
+        isConfirmPasswordFocused = confirmPasswordFocusNode.hasFocus;
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    emailFocusNode.dispose();
+    passwordFocusNode.dispose();
+    confirmPasswordFocusNode.dispose();
+    super.dispose();
+  }
+
+  bool isKeyboardOpen(BuildContext context) {
+    return MediaQuery.of(context).viewInsets.bottom != 0;
+  }
+
+  void handelSubmit() {
+    bool someNotFilled = emailController.text.isEmpty ||
+        passwordController.text.isEmpty ||
+        confirmPasswordController.text.isEmpty;
+
+    if (emailController.text.isEmpty) {
+      emailShakeKey.currentState?.shake();
+    } else if (passwordController.text.isEmpty) {
+      passwordShakeKey.currentState?.shake();
+    } else if (confirmPasswordController.text.isEmpty) {
+      confirmPasswordShakeKey.currentState?.shake();
+    }
+
+    if (someNotFilled) {
+      Vibration.hasVibrator().then((hasVibrator) {
+        if (hasVibrator ?? false) {
+          Vibration.vibrate(duration: 100);
+        }
+      });
+    }
+  }
+
+  String? customValidation(String? value) {
+    return confirmPasswordValidation(passwordController.text, value);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Palette.background,
+      body: Padding(
+        padding: const EdgeInsets.only(top: 40),
+        child: Form(
+            key: formKey,
+            child: Stack(
+              children: [
+                Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: <Widget>[
+                    SizedBox(
+                      height: isKeyboardOpen(context) ? 160 : 0,
+                    ),
+                    const TitleElement(
+                        name: 'Your email address',
```

Comment:
I know it is in the design, but I should it should be changed to Sign Up.
talk to us on the group about it.